code/AssetLib/VRML/contrib/meshlab/src/meshlabplugins/io_x3d/vrml/Scanner.cpp

```
@@ -0,0 +1,931 @@
+	keywords.set(L"TRUE", 82);
+	keywords.set(L"FALSE", 83);
+	keywords.set(L"SALVE", 84);
+
+
+	tvalLength = 128;
+	tval = new wchar_t[tvalLength]; // text of current token
+
+	// HEAP_BLOCK_SIZE byte heap + pointer to next heap block
+	heap = malloc(HEAP_BLOCK_SIZE + sizeof(void*));
+	firstHeap = heap;
+	heapEnd = (void**) (((char*) heap) + HEAP_BLOCK_SIZE);
```

Cppcheck (2.19.0)

[error] If memory allocation fails

(nullPointerArithmeticOutOfMemory)